### ISSUE TITLE
exec("/data/data/com.termux/files/usr/bin/login"): Permission denied  [Process completed (code 1) - press Enter]

### ISSUE THREAD

**Issue by dpnkrmaurya** [termux/termux-app#2456]
### Problem description

Show message: exec("/data/data/com.termux/files/usr/bin/login"): Permission denied

[Process completed (code 1) - press Enter]
![Screenshot_2021-12-07-02-20-01](https://user-images.githubusercontent.com/58885577/145049664-fea52bcc-d829-436c-b5f0-9da951d3958a.png)

### Steps to reproduce the behavior.

exec("/data/data/com.termux/files/usr/bin/login"): Permission denied

[Process completed (code 1) - press Enter]

### What is the expected behavior?

Error 

### System information

* Termux application version: 0.117
* Android OS version: 7
* Device model: Micromax Bharat 5 pro

**Comment by Grimler91** [comment]
Works fine here. How/from where did you install termux?

**Comment by ieyhan09** [comment]
![Screenshot_2021-12-07-21-52-02-66_84d3000e3f4017145260f7618db1d683.jpg](https://user-images.githubusercontent.com/75145549/145055686-b3892b0d-0e01-4601-b92e-a4cbc02912da.jpg)

**Comment by dpnkrmaurya** [comment]
F-Droid source

**Comment by Luckymlucky** [comment]
Ok

**Comment by Luckymlucky** [comment]
@Luckymlucky

**Comment by Luckymlucky** [comment]
#
Duplicate of #

**Comment by 2096779623** [comment]
Right slip,NEW SESSIONS,Click First button,I think it's called "safe mode", You can try chmod or rm the /data/data/com.termux/files/usr/bin/login

**Comment by Luckymlucky** [comment]
That link was  directory
How to put the command
i did not access it
So please accept it the mail and give me helpful tool solve it.

**Comment by Practicalbutterfly5** [comment]
I ran into same issue after restoring my migrate backup.

Solution was pretty simple for me.

1. Open termux in failsafe mode "Long press termux icon>failsafe".

2. Backup termux-data to sdcard with
`/sbin/su -c "tar -zcf /sdcard/termux-backup.tar.gz -C /data/data/com.termux/files ./home ./usr"`

3. Clear termux data.

4. Open termux and type `termux-setup-storage` and `tar -zxf /sdcard/termux-backup.tar.gz -C /data/data/com.termux/files --recursive-unlink --preserve-permissions`.  Restart termux and you should be good.

**Comment by WyroxGai** [comment]
​#!​/bin/bash 
  
 ​detect_distro​() { 
 ​    ​if​ [[ ​"​$OSTYPE​"​ ​==​ linux-android​*​ ]]​;​ ​then 
 ​            distro=​"​termux​" 
 ​    ​fi 
  
 ​    ​if​ [ ​-z​ ​"​$distro​"​ ]​;​ ​then 
 ​        distro=​$(​ls /etc ​|​ awk ​'​match($0, "(.+?)[-_](?:release|version)", groups) {if(groups[1] != "os") {print groups[1]}}​'​) 
 ​    ​fi 
  
 ​    ​if​ [ ​-z​ ​"​$distro​"​ ]​;​ ​then 
 ​        ​if​ [ ​-f​ ​"​/etc/os-release​"​ ]​;​ ​then 
 ​            distro=​"​$(​source /etc/os-release ​&&​ ​echo​ ​$ID​)​" 
 ​        ​elif​ [ ​"​$OSTYPE​"​ ​==​ ​"​darwin​"​ ]​;​ ​then 
 ​            distro=​"​darwin​" 
 ​        ​else​  
 ​            distro=​"​invalid​" 
 ​        ​fi 
 ​    ​fi 
 ​} 
  
 ​pause​() { 
 ​    ​read​ -n1 -r -p ​"​Press any key to continue...​"​ key 
 ​} 
 ​banner​() { 
 ​    clear 
 ​    ​echo​ -e ​"​\e[1;31m​" 
 ​    ​if​ ​!​ [ ​-x​ ​"​$(​command -v figlet​)​"​ ]​;​ ​then 
 ​        ​echo​ ​'​Introducing TBomb​' 
 ​    ​else 
 ​        figlet TBomb 
 ​    ​fi 
 ​    ​if​ ​!​ [ ​-x​ ​"​$(​command -v toilet​)​"​ ]​;​ ​then 
 ​        ​echo​ -e ​"​\e[4;34m This Bomber Was Created By \e[1;32mSpeedX \e[0m​" 
 ​    ​else 
 ​        ​echo​ -e ​"​\e[1;34mCreated By \e[1;34m​" 
 ​        toilet -f mono12 -F border SpeedX 
 ​    ​fi 
 ​    ​echo​ -e ​"​\e[1;34m For Any Queries Join Me!!!\e[0m​" 
 ​    ​echo​ -e ​"​\e[1;32m           Telegram: https://t.me/TBombChat \e[0m​" 
 ​    ​echo​ -e ​"​\e[4;32m   YouTube: https://www.youtube.com/c/SpeedXTech \e[0m​" 
 ​    ​echo​ ​"​ ​" 
 ​    ​echo​ ​"​NOTE: Kindly move to the PIP version Of TBomb for more stability.​" 
 ​    ​echo​ ​"​ ​" 
 ​} 
  
 ​init_environ​(){ 
 ​    ​declare​ -A backends​;​ backends=( 
 ​        [​"​arch​"​]=​"​pacman -S --noconfirm​" 
 ​        [​"​debian​"​]=​"​apt-get -y install​" 
 ​        [​"​ubuntu​"​]=​"​apt -y install​" 
 ​        [​"​termux​"​]=​"​apt -y install​" 
 ​        [​"​fedora​"​]=​"​yum -y install​" 
 ​        [​"​redhat​"​]=​"​yum -y install​" 
 ​        [​"​SuSE​"​]=​"​zypper -n install​" 
 ​        [​"​sles​"​]=​"​zypper -n install​" 
 ​        [​"​darwin​"​]=​"​brew install​" 
 ​        [​"​alpine​"​]=​"​apk add​" 
 ​    ) 
  
 ​    INSTALL=​"​${backends[$distro]}​" 
  
 ​    ​if​ [ ​"​$distro​"​ ​==​ ​"​termux​"​ ]​;​ ​then 
 ​        PYTHON=​"​python​" 
 ​        SUDO=​"​" 
 ​    ​else 
 ​        PYTHON=​"​python3​" 
 ​        SUDO=​"​sudo​" 
 ​    ​fi 
 ​    PIP=​"​$PYTHON​ -m pip​" 
 ​} 
  
 ​install_deps​(){ 
 ​     
 ​    packages=(openssl git ​$PYTHON​ ​$PYTHON​-pip figlet toilet) 
 ​    ​if​ [ ​-n​ ​"​$INSTALL​"​ ]​;​then 
 ​        ​for​ ​package​ ​in​ ​${packages[@]}​;​ ​do 
 ​            ​$SUDO​ ​$INSTALL​ ​$package 
 ​        ​done 
 ​        ​$PIP​ install -r requirements.txt 
 ​    ​else 
 ​        ​echo​ ​"​We could not install dependencies.​" 
 ​        ​echo​ ​"​Please make sure you have git, python3, pip3 and requirements installed.​" 
 ​        ​echo​ ​"​Then you can execute bomber.py .​" 
 ​        ​exit 
 ​    ​fi 
 ​} 
  
 ​banner 
 ​pause 
 ​detect_distro 
 ​init_environ 
 ​if​ [ ​-f​ .update ]​;​then 
 ​    ​echo​ ​"​All Requirements Found....​" 
 ​else 
 ​    ​echo​ ​'​Installing Requirements....​' 
 ​    ​echo​ ​. 
 ​    ​echo​ ​. 
 ​    install_deps 
 ​    ​echo​ This Script Was Made By SpeedX ​>​ .update 
 ​    ​echo​ ​'​Requirements Installed....​' 
 ​    pause 
 ​fi 
 ​while​ ​: 
 ​do 
 ​    banner 
 ​    ​echo​ -e ​"​\e[4;31m Please Read Instruction Carefully !!! \e[0m​" 
 ​    ​echo​ ​"​ ​" 
 ​    ​echo​ ​"​Press 1 To  Start SMS  Bomber ​" 
 ​    ​echo​ ​"​Press 2 To  Start CALL Bomber ​" 
 ​    ​echo​ ​"​Press 3 To  Start MAIL Bomber (Not Yet Available)​" 
 ​    ​echo​ ​"​Press 4 To  Update (Works On Linux And Linux Emulators) ​" 
 ​    ​echo​ ​"​Press 5 To  Exit ​" 
 ​    ​read​ ch 
 ​    clear 
 ​    ​if​ [ ​$ch​ ​-eq​ 1 ]​;​then 
 ​        ​$PYTHON​ bomber.py --sms 
 ​        ​exit 
 ​    ​elif​ [ ​$ch​ ​-eq​ 2 ]​;​then 
 ​        ​$PYTHON​ bomber.py --call 
 ​        ​exit 
 ​    ​elif​ [ ​$ch​ ​-eq​ 3 ]​;​then 
 ​        ​$PYTHON​ bomber.py --mail 
 ​        ​exit 
 ​    ​elif​ [ ​$ch​ ​-eq​ 4 ]​;​then 
 ​        ​echo​ -e ​"​\e[1;34m Downloading Latest Files...​" 
 ​        rm -f .update 
 ​        ​$PYTHON​ bomber.py --update 
 ​        ​echo​ -e ​"​\e[1;34m RUN TBomb Again...​" 
 ​        pause 
 ​        ​exit 
 ​    ​elif​ [ ​$ch​ ​-eq​ 5 ]​;​then 
 ​        banner 
 ​        ​exit 
 ​    ​else 
 ​        ​echo​ -e ​"​\e[4;32m Invalid Input !!! \e[0m​" 
 ​        pause 
 ​    ​fi 
 ​done

**Comment by agnostic-apollo** [comment]
Install `v0.118.0` and check again. If crash happens, post output.

**Comment by twaik** [comment]
Looks like this issue can be closed...

**Comment by Dumfffg** [comment]
## Transcript

```
exec("/data/data/com.termux/files/usr/bin/login"): Permission denied

[Process completed (code 1) - press Enter]
```
##


## Termux App Info

**APP_NAME**: `Termux`  
**PACKAGE_NAME**: `com.termux`  
**VERSION_NAME**: `0.118.0`  
**VERSION_CODE**: `118`  
**TARGET_SDK**: `28`  
**IS_DEBUGGABLE_BUILD**: `false`  
**APK_RELEASE**: `F-Droid`  
**SIGNING_CERTIFICATE_SHA256_DIGEST**: `228FB2CFE90831C1499EC3CCAF61E96E8E1CE70766B9474672CE427334D41C42`  
##


## Device Info

### Software

**OS_VERSION**: `4.4.22+`  
**SDK_INT**: `25`  
**RELEASE**: `7.1.1`  
**ID**: `N6F26Q`  
**DISPLAY**: `CPH1729EX_11_A.35_200903`  
**INCREMENTAL**: `1599151965`  
**SECURITY_PATCH**: `2019-10-05`  
**IS_DEBUGGABLE**: `0`  
**TYPE**: `user`  
**TAGS**: `release-keys`  

### Hardware

**MANUFACTURER**: `OPPO`  
**BRAND**: `OPPO`  
**MODEL**: `CPH1729`  
**PRODUCT**: `CPH1729`  
**BOARD**: `full_oppo6763_17101`  
**HARDWARE**: `mt6763`  
**DEVICE**: `CPH1729`  
**SUPPORTED_ABIS**: `arm64-v8a, armeabi-v7a, armeabi`  
##

**Comment by 2096779623** [comment]
> ## Transcript
> ```
> exec("/data/data/com.termux/files/usr/bin/login"): Permission denied
> 
> [Process completed (code 1) - press Enter]
> ```

Go to `failsafe` first: https://wiki.termux.com/wiki/Recover_a_broken_environment
Then execute: `stats /data/data/com.termux/files/usr/bin/login`

**Comment by SowmiyanS** [comment]
I tried all the commands, but what worked for me was to factory reset my phone and then installing the latest version of termux.

**Comment by TechnicalStudioDeveloper** [comment]
Same problem here too
Device : 
Android  -10
Redmi 9

**Comment by gamecss** [comment]
![Screenshot_20230903-100159_Termux](https://github.com/termux/termux-app/assets/52902973/02606a35-e1aa-4047-9c01-cad4629a8ab9)
Same issue. Happened after remove /data/app/com.termux* and /data/data/com.termux* in recovery manually to resolve boot loop

**Comment by gamecss** [comment]
> ![Screenshot_20230903-100159_Termux](https://user-images.githubusercontent.com/52902973/265238189-02606a35-e1aa-4047-9c01-cad4629a8ab9.png) Same issue. Happened after remove /data/app/com.termux* and /data/data/com.termux* in recovery manually to resolve boot loop

Though I change login's permission to 777, it's still unexecutable

**Comment by agnostic-apollo** [comment]
You have `untrusted_app` source context assigned to the termux app process, it should be `untrusted_app_27`. Try rebooting phone.

**Comment by gamecss** [comment]
> You have `untrusted_app` source context assigned to the termux app process, it should be `untrusted_app_27`. Try rebooting phone.

~It still doesn't work.~
After format userdata, everything works well

**Comment by FalconSN** [comment]
Old issue, but I'd like to add my experience with this problem. I ran into the same issue when I restored my backup from tar file. The problem was strangely symlinks. All files were fine except for them. `ls $PREFIX/lib` could see all files in there except for symlinks, they threw permission denied error despite being owned by Termux and with correct permissions, hence bash couldn't load the required library files. Running `sh -x /data/data/com.termux/files/usr/bin/login` in failsafe mode helped with noticing this. Removing those symlinks as root and linking again as regular user fixed it.
